mux_test.go

```diff
@@ -405,16 +405,6 @@ func TestPath(t *testing.T) {
 			pathTemplate: `/{type:(?i:daily|mini|variety)}-{date:\d{4,4}-\d{2,2}-\d{2,2}}`,
 			shouldMatch:  true,
 		},
-		{
```

Comment:
@elithrar Yes, breaking this test means that `extractVars` is showing the faulty behavior again, which is not surprising since this PR reverts `extractVars` to its earlier implementation.